### PR TITLE
monitoring: Update dashboards

### LINKED ISSUE #18124: Clean up code intels grafana dashboards
We have some unused panels that could be removed and metrics that seem to be deprecated.

### LINKED ISSUE #21606: codeintel: Add dashboards for dependency queue
Following up from action items listed in https://github.com/sourcegraph/sourcegraph/pull/21476.

### LINKED ISSUE #21644: codeintel: Add dashboard for executor
We need to write a dashboard for executor metrics (not just the executor-queue). We also need to ensure that we can scrape the metrics from the executors in the sibling GCP project for Cloud.

### PR DESCRIPTION
**Pull request by efritz**
This PR updates the following dashboards to use the functions introduced in #23054.

- executor-queue
- executor (previously precise-code-intel-indexer)
- frontend (partial)
- precise-code-intel-worker
- worker

Fixes #18124. Fixes #21606. Fixes #21644.

---

Frontend (default view):
Screen Shot 2021-07-23 at 3 03 01 PM

Frontend (expanded):
Screen Shot 2021-07-23 at 3 02 00 PM

---

Worker (default view):
Screen Shot 2021-07-23 at 3 03 03 PM

Worker (expanded):
Screen Shot 2021-07-23 at 3 09 55 PM

Worker (expanded, 2):
Screen Shot 2021-07-23 at 3 02 23 PM

Worker (expanded, 3):
Screen Shot 2021-07-23 at 3 02 26 PM

---

Precise code intel worker (default view):
Screen Shot 2021-07-23 at 3 03 05 PM

Precise code intel worker (expanded):
Screen Shot 2021-07-23 at 3 02 40 PM

Precise code intel worker (expanded, 2):
Screen Shot 2021-07-23 at 3 02 42 PM

---

Executor (default view):
Screen Shot 2021-07-23 at 3 03 08 PM

Executor (expanded):
Screen Shot 2021-07-23 at 3 02 50 PM

---

Executor queue (default view):
Screen Shot 2021-07-23 at 3 03 10 PM

Executor queue (expanded):
Screen Shot 2021-07-23 at 3 02 54 PM

### CODE REVIEW

**Review by Strum355 — APPROVED** [review]

**Comment by efritz** [comment]
Note: I've stacked https://github.com/sourcegraph/sourcegraph/pull/23194 on top of this which reorganizes the new panel definitions. They may be easier to read in that state as they are not per-service but just defined as a common panel constructor taking a container name.

File: monitoring/definitions/shared/standard.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func (standardConstructor) Errors(legend string) observableConstructor {
 			return Observable{
 				Name:        fmt.Sprintf("%s_errors_total", options.MetricNameRoot),
 				Description: fmt.Sprintf("%s%s errors every 5m", options.MetricDescriptionRoot, legend),
-				Query:       fmt.Sprintf(`sum%s(increase(src_%s_errors_total{%s}[5m]))`, by, options.MetricNameRoot, filters),
+				Query:       fmt.Sprintf(`sum%s(increase(src_%s_errors_total{%s}[5m])) or on() vector(0)`, by, options.MetricNameRoot, filters),
 				Panel:       monitoring.Panel().LegendFormat(fmt.Sprintf("%s%s errors", legendPrefix, legend)),
```
Comment (bobheadxi) [inline monitoring/definitions/shared/standard.go:90]: Might I recommend using our panel options library? 😁 This way, we can still have a way of knowing the difference between 0 and missing, but also have nicer cosmetics on the graphs:

```suggestion
				Query:       fmt.Sprintf(`sum%s(increase(src_%s_errors_total{%s}[5m]))`, by, options.MetricNameRoot, filters),
				Panel:       monitoring.Panel().LegendFormat(fmt.Sprintf("%s%s errors", legendPrefix, legend))..With(monitoring.PanelOptions.ZeroIfNoData()),
```

**Comment by bobheadxi** [comment]
Also @efritz not sure if you are aware but if you want more real-looking data when testing out panels, I wrote a guide on pointing local grafana at another Sourcegraph here: https://docs.sourcegraph.com/dev/how-to/monitoring_local_dev#grafana 

(suggesting bc screenshots look like you are spinning up sourcegraph locally to test :) )

**Comment by efritz** [comment]
@bobheadxi Cool! This wouldn't have worked for this effort since I was also altering what metrics were sent from the backend. But this will be super nice in the future when I'm tweaking alerts and adding new dashboards (after a metrics deploy).

File: monitoring/definitions/shared/standard.go (hunk 1 above)
Comment (efritz) [reply monitoring/definitions/shared/standard.go]: I swear I used this _incorrectly_ in the past and it didn't work

**Review by bobheadxi — APPROVED** [review]
very nice

**Pull request merged**